src/libs/actions/Policy.js

```diff
@@ -131,6 +131,41 @@ function createAndNavigate(name = '') {
     create(name).then(navigateToPolicy);
 }
 
+
+/**
+ * Delete the policy
+ *
+ * @param {String} [policyID]
+ * @param {Boolean} shouldGrowl
+ * @param {Boolean} shouldAutomaticallyReroute
+ * @returns {Promise}
+ */
+function deletePolicy(policyID = '', shouldGrowl = true, shouldAutomaticallyReroute = true) {
+    return API.Policy_Delete({policyID})
+        .then((response) => {
+            if (response.jsonCode !== 200) {
+                // Show the user feedback
+                const errorMessage = translateLocal('workspace.new.genericFailureMessage');
+                Growl.error(errorMessage, 5000);
+                return;
+            }
+
+            if (shouldGrowl) {
+                Growl.show(translateLocal('workspace.common.growlMessageOnDelete'), CONST.GROWL.SUCCESS, 3000);
+            }
+
+            // Removing the workspace data from Onyx as well
+            return Onyx.set(`${ONYXKEYS.COLLECTION.POLICY}${policyID}`, null);
```

Comment:
i believe you are missing the underscore `_` between your policy and policyID thats why the actual onyx variable is not getting cleared which is why i can still see the workspace after deleting it.